src/Sulu/Bundle/MediaBundle/DependencyInjection/ImageFormatCompilerPass.php

```diff
@@ -54,20 +54,25 @@ protected function loadThemeFormats($defaultOptions)
 
         $activeTheme = $this->container->get('liip_theme.active_theme');
         $bundles = $this->container->getParameter('kernel.bundles');
+        $asseticBundles = $this->container->getParameterBag()->resolveValue($this->container->getParameter('assetic.bundles'));
+        $configFolderFiles = $this->container->getParameter('sulu_media.format_manager.config_files');
+        $defaultFolder = '/config/image-formats.xml';
 
         foreach ($activeTheme->getThemes() as $theme) {
-            foreach ($bundles as $bundle => $class) {
-                $reflector = new \ReflectionClass($class);
+            foreach ($asseticBundles as $bundleName) {
+                $reflector = new \ReflectionClass($bundles[$bundleName]);
+                $configPath = $defaultFolder;
+                if (isset($configFolderFiles[$theme])) {
+                    $configPath = $configFolderFiles[$theme];
+                }
+                $fullPath = dirname($reflector->getFileName())
+                    . '/Resources/themes/' . $theme . '/' . ltrim($configPath, '/');;
+
                 if ($reflector->getFileName() &&
```

Comment:
In what situation would the reflector not have a filename? Each reflector is a bundle and each bundle has a filename .. or ?